### PR TITLE
[plugin.program.rtorrent] 19.21.10

### PR DESCRIPTION
**Pull request by djcode**
### Description
A simple plugin for interacting with an rTorrent instance.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by djcode** [comment]
@enen92 Is there anything else I need to do to have this plugin merged in?

**Comment by enen92** [comment]
Someone to actually review the code, I haven't looked yet.

File: plugin.program.rtorrent/resources/lib/functions.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return match.group(1) if match else name
+
+def play_file(url):
+    ''' Function to determine whether to play video/audio or show an image '''
+    mimetype = mimetypes.guess_type(url)[0]
+    if mimetype and any(re.findall(r'image|audio|video', mimetype)):
+        if 'image' in mimetype:
+            xbmc.executebuiltin('ShowPicture(%s)' % url)
+        elif 'video' in mimetype or 'audio' in mimetype:
+            xbmc.Player().play(url)
```
Comment (basrieter) [inline plugin.program.rtorrent/resources/lib/functions.py:85]: You should not call the `xbmc.Player().play(url)` but use the `xbmcplugin.setResolvedUrl(handle, succeeded, listitem)`. See https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/group__python__xbmcplugin.html#gaa03ad528583db7d5277eae3977acde0f

**Review by basrieter — CHANGES_REQUESTED** [review]
See my comments.

**Comment by enen92** [comment]
ping @djcode

**Comment by djcode** [comment]
Thanks @enen92. I missed the notification that this had been looked at.

Trying to wrap my head on how I can accomplish the requested changes. I'm calculating the file path and availability on playback and if I'm missing something obvious, the listitem object would not be available to me at that point. For example, If the file is not completely downloaded, I issue a confirmation if the user is sure they want to play it. In the past, end users trying to play incomplete files has caused Kodi to lock up, so I figure the warning is needed.

Can I pass a listitem object between RunPlugin calls?

**Comment by enen92** [comment]
Closing as stalled, please resubmit if still interested

**Pull request closed without merging**